packages/front-end/components/Metrics/MetricForm/index.tsx

```diff
@@ -427,6 +444,37 @@ const MetricForm: FC<MetricFormProps> = ({
     }
   }, [type, form]);
 
+  const tableOptions: { label: string; value: string }[] = [];
+
+  const { data: TableData } = useApi<{
+    informationSchema: InformationSchemaInterface;
+  }>(`/datasource/${selectedDataSource.id}/schema`);
+
+  if (TableData?.informationSchema?.databases.length) {
+    TableData.informationSchema.databases.forEach((database) => {
+      database.schemas.forEach((schema) => {
+        schema.tables.forEach((table) => {
+          tableOptions.push({ label: table.tableName, value: table.id });
+        });
+      });
+    });
+  }
+
+  const columnOptions: { label: string; value: string }[] = [];
+
+  const { data: columnData } = useApi<{
+    table: InformationSchemaTablesInterface;
+  }>(`/datasource/${selectedDataSource.id}/schema/table/${tableId}`);
```

Comment:
This will make an API call even if tableId is empty.

Reply:
@jdorn Do you have any pattern suggestions to get around this since you can't call a React hook conditionally? Currently, if a tableId or columnId is empty, the call just returns a 404.

Reply:
Our `useApi` hook is just a thin wrapper around the useSWR library.  They have a way to do conditional fetching - https://swr.vercel.app/docs/conditional-fetching#conditional

To support this, we would need to modify our `useApi` hook a little bit.
1. Allow the `path` argument to be null
2. If `path` is null, use `null` as the key instead of `orgId + "::" + path`

After that, we can use it like this:
```ts
useApi(
  tableId ? `/datasource/${selectedDataSource.id}/schema/table/${tableId}` : null
)
```

Reply:
@jdorn Thank you!